cmd/juju/cloud/removecredential.go

```diff
@@ -127,22 +116,14 @@ func (c *removeCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
 }
 
 func (c *removeCredentialCommand) Run(ctxt *cmd.Context) error {
-	if c.BothClientAndController || c.ControllerOnly {
-		if c.ControllerName == "" {
-			// The user may have specified the controller via a --controller option.
-			// If not, let's see if there is a current controller that can be detected.
-			var err error
-			c.ControllerName, err = c.MaybePromptCurrentController(ctxt, fmt.Sprintf("remove credential %q for cloud %q from", c.credential, c.cloud))
-			if err != nil {
-				return errors.Trace(err)
-			}
-		}
+	if err := c.MaybePrompt(ctxt, fmt.Sprintf("remove credential %q for cloud %q from", c.credential, c.cloud)); err != nil {
+		return errors.Trace(err)
 	}
-	if c.ControllerName == "" && !c.ClientOnly {
-		ctxt.Infof("To remove credential %q for cloud %q from this client, use the --client-only option.", c.credential, c.cloud)
+	if c.ControllerName == "" && !c.Client {
+		ctxt.Infof("To remove credential %q for cloud %q from this client, use the --client option.", c.credential, c.cloud)
 	}
 	var client RemoveCredentialAPI
-	if !c.ClientOnly {
+	if c.ControllerName != "" {
```

Comment:
This logic can be moved down to the other if controllerName != "" block

Reply:
No, it can't since the line right after it uses this client. We are checking remote clouds and needed to open that connection.